build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -157,7 +158,22 @@ private static void RemovePdbsFromDir(string directory)
             FS.RmFilesInDirRecursive(directory, "*.pdb");
         }
 
-        private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli dotnet, string rootOutputDirectory)
+        private static BuildTargetResult CompileCliSdkAndGenerateNuGetPackagesArchive(
+            BuildTargetContext c,
+            DotNetCli dotnet,
+            string rootOutputDirectory,
+            DotNetCli currentDotnet)
```

Comment:
I don't understand the need for both "DotNetCli" parameters.  This method is only called in 1 spot, and the same value is passed in for both variables.  I think we only need 1.  And then we call:

`CompileCliSdk(c, dotnet, rootOutputDirectory, dotnet);`